packages/noya-state/src/reducers/application.ts

```diff
@@ -853,6 +871,116 @@ export function reducer(
         }
       });
     }
+    case 'addColorSwatch': {
+      return produce(state, (state) => {
+        const sharedSwatches = state.sketch.document.sharedSwatches;
+
+        const swatchColor: Sketch.Color = {
+          "_class": "color",
+          "alpha": 0.5,
+          "red": 0,
+          "green": 0,
+          "blue": 0
+        }
+        const swatch: Sketch.Swatch = {
+          "_class": "swatch",
+          "do_objectID": uuid(),
+          "name": "Undefined",
+          "value": swatchColor,
+        }
+
+        sharedSwatches?.objects.push(swatch);
```

Comment:
We need to remove the `?` so that this still works if there are no swatches yet. Maybe we can use the `getSharedSwatches` selector? (once we update it)

Reply:
I modified it to create it in case it doesn't exist.